apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibility.jsx

```diff
@@ -1,51 +1,127 @@
 import {FormGroup, Button} from 'react-bootstrap';
 import FieldGroup from '../../code-studio/pd/form_components/FieldGroup';
 import SchoolAutocompleteDropdownWithLabel from '@cdo/apps/templates/census2017/SchoolAutocompleteDropdownWithLabel';
 import AmazonFutureEngineerEligibilityForm from './amazonFutureEngineerEligibilityForm';
 import AmazonFutureEngineerAccountConfirmation from './amazonFutureEngineerAccountConfirmation';
+import {pegasus} from '@cdo/apps/lib/util/urlHelpers';
 
 const styles = {
   intro: {
     paddingBottom: 10
   }
 };
 
+const sessionStorageKey = 'AmazonFutureEngineerEligibility';
+
 export default class AmazonFutureEngineerEligibility extends React.Component {
+  static propTypes = {
+    signedIn: PropTypes.bool.isRequired
+  };
+
   constructor(props) {
     super(props);
 
+    let sessionEligibilityData =
+      JSON.parse(sessionStorage.getItem(sessionStorageKey)) || {};
+
     this.state = {
-      schoolEligible: null,
-      consent: false,
-      email: ''
+      formData: {
+        signedIn: this.props.signedIn,
+        schoolEligible:
+          'schoolEligible' in sessionEligibilityData
+            ? sessionEligibilityData.schoolEligible
+            : null,
+        consentAFE:
+          'consentAFE' in sessionEligibilityData
+            ? sessionEligibilityData.consentAFE
+            : false
+      }
     };
   }
 
-  handleChange = change => {
-    this.setState(change);
+  updateFormData = change => {
+    this.setState({formData: {...this.state.formData, ...change}});
   };
 
-  submit = () => {
-    // TO DO: actually check whether a school is eligible
+  saveToSessionStorage = () => {
+    sessionStorage.setItem(
+      sessionStorageKey,
+      JSON.stringify(this.state.formData)
+    );
+  };
+
+  handleClickCheckEligibility = () => {
     // TO DO: if ineligible, open new ineligibility page (markdown that marketing can edit)
-    this.setState({
-      schoolEligible: true
+
+    if (this.state.formData.schoolId === '-1') {
+      this.handleEligibility(false);
+    }
+
+    $.ajax({
+      url: '/dashboardapi/v1/schools/' + this.state.formData.schoolId,
+      type: 'get',
+      dataType: 'json'
+    }).done(schoolData => {
+      this.handleEligibility(schoolData.afe_high_needs);
     });
   };
 
-  handleSchoolDropdownChange = (field, event) => {
+  handleEligibility(isEligible) {
     this.setState({
+      formData: {...this.state.formData, ...{schoolEligible: isEligible}}
+    });
+    this.saveToSessionStorage();
+
+    if (!isEligible) {
+      window.location = pegasus('/privacy');
+    }
+  }
+
+  handleSchoolDropdownChange = (field, event) => {
+    const newData = {
       schoolId: event ? event.value : '',
       schoolName: event ? event.label : ''
-    });
+    };
+
+    this.updateFormData(newData);
+  };
+
+  loadConfirmationPage = () => {
+    this.saveToSessionStorage();
+
+    return <AmazonFutureEngineerAccountConfirmation />;
+  };
+
+  loadCompletionPage = () => {
```

Comment:
@molly-moen something for next week...realizing this may be a problem with your work. 

I'm currently setting things up so that the API request to AFE is made in the `amazonFutureEngineerEligibility` component, which I think you would not be using in (you'd go straight to the `AmazonFutureEngineerForm` component, yeah)? Let's figure out what the best approach would be...

Reply:
Another thought is we could set up `AmazonFutureEngineerEligibility` to do an initial check of a user's email/school information if they're logged in, and then show the form if they've already met the school requirement (which it seems like people coming from your banner would).

Reply:
yeah the page will only be shown if they are eligible from my side

Reply:
would there be any issues with moving the api request to this component instead of the outer component?

Reply:
Chatted about this offline -- plan is to modify initial API request to check whether a user is signed in to also get their school information. If their school is eligible, we'll skip to the second step (form to consent / give shipping information).